torch/csrc/api/include/torch/data/datasets/chunk.h

```diff
@@ -415,14 +552,15 @@ class ChunkDataset final
   // batches and caches them in batch_buffer_.
   ChunkReader chunk_reader_;
 
-  // chunk sampler to shuffle different chunks
-  samplers::LockedSampler<ChunkSamplerType> chunk_sampler_;
-
   // example sampler to shuffle examples in a specific chunk
   ExampleSamplerType example_sampler_;
 
+  // Selects chunks and their order for this reader.
+  std::shared_ptr<detail::ChunkSelector> chunk_selector_;
```

Comment:
It looks like ChunkDataset uniquely own this selector. Shall we make it a unique_ptr?